src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceDeserializer.cs

```diff
@@ -225,42 +225,48 @@ internal ODataDeletedResource ReadDeletedEntry()
             Uri id = null;
             DeltaDeletedEntryReason reason = DeltaDeletedEntryReason.Changed;
 
-            // If the current node is the id property - read it.
-            if (this.JsonReader.NodeType == JsonNodeType.Property &&
-                string.Equals(JsonLightConstants.ODataIdPropertyName, this.JsonReader.GetPropertyName(), StringComparison.Ordinal))
+            do

```

Comment:
Typically, for property reading, we are using  `while () {}` loop.
I am not fans for 'while()' loop. But, If we can keep it consistent, it could be better.

Reply:
@xuzhg I replaced the `do {} while` into a `while {}` loop. I figure it especially made sense if the JSON was malformed